vpr/src/base/partition_regions.h

```diff
@@ -0,0 +1,37 @@
+#ifndef PARTITION_REGIONS_H
+#define PARTITION_REGIONS_H
+
+/**
+ * @file
+ * @brief This file defines the PartitionRegions class. The PartitionRegions class is used to store the union
+ * of regions that a particular partition can be placed in.
+ */
+
+class PartitionRegions {
+  public:
```

Comment:
add a simple constructor - takes no arguments and sets everything to empty.
comment every member function, or group them together and comment accordingly